lib/vyasa/medium/store.ex

```diff
@@ -0,0 +1,97 @@
+defmodule Vyasa.Medium.Store do
+  @moduledoc """
+  S3 Object Storage Service Communicator  using HTTP POST sigv4
+  https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-post-example.html
+  """
+
+  alias Vyasa.Medium
+
+  @bucket "vyasa"
+
+  def path(st) when is_struct(st) do
+    #input and output paths
+    {local_path(st), path_constructor(st)}
+  end
+
+  def get(st) when is_struct(st) do
+    signer(:get, path_constructor(st))
+  end
+
+  def get(path) do
+    signer(:get, path)
+  end
+
+  def get!(st) when is_struct(st) do
+    signer!(:get, path_constructor(st))
+  end
+
+  def get!(path) do
+    signer!(:get, path)
+  end
+
+  def put(struct) do
+    signer(:put, path_constructor(struct))
+  end
+
+  def put!(struct) do
+    signer!(:put, path_constructor(struct))
+  end
+
+  def hydrate(%Medium.Voice{} = voice) do
```

Comment:
voice can be dehydrated most times 

only needs to be hydrated when playback creation required -- uses get only